### ISSUE TITLE
Add JSON schema to the `typescript` package for `tsconfig.json` and `jsconfig.json`

### ISSUE THREAD

**Issue by scottwillmoore** [microsoft/TypeScript#54192]
# Suggestion

## 🔍 Search Terms

- JSON schema
- SchemaStore
- `tsconfig.json`
- `jsconfig.json`

## ✅ Viability Checklist

N/A

## 📃 Why

I noticed that [Rome](https://rome.tools/) doesn't use a traditional hosted JSON schema and instead they publish their JSON schema in their package so that it can then be referenced with a local path:

```json
{
  "$schema": "./node_modules/rome/configuration_schema.json",
}
```

I see three advantages to this approach...

1. The JSON schema will match the installed version of the package.
2. The JSON schema will work without internet access, provided that node_modules has been installed.
3. Any changes to the JSON schema in the development can be used straight away in examples, etc.

## ⭐ Suggestion

Maintain the JSON schema in this repository, instead of the the [`SchemaStore`](https://github.com/SchemaStore/schemastore) repository. It will be kept in sync with new TypeScript releases. Finally, publish the JSON schemas in the `typescript` package.

You can still leave the hosted JSON schema as an option, which will always resolve to the JSON schemas for the latest version of TypeScript. This will still be needed for editor autocomplete, whereby the schema is not specified in the JSON, but detected from the name of the file.

**Comment by terenc3** [comment]
I agree. A local schema file is better. compodoc is doing it also this way, see [[FEATURE] Provide a JSON Schema for JSON Configuration File. #577](https://github.com/compodoc/compodoc/issues/577)

**Comment by jonasgeiler** [comment]
Biome.js has also started doing this and it's very convenient: https://biomejs.dev/reference/configuration/#schema

**Comment by XantreDev** [comment]
Should we open a PR with a copy of json file from `https://json.schemastore.org/tsconfig` xD?
